be/src/exec/pipeline/scan/olap_meta_chunk_source.h

```diff
@@ -0,0 +1,38 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+#pragma once
+
+#include "column/vectorized_fwd.h"
+#include "exec/pipeline/scan/chunk_source.h"
+#include "exec/pipeline/scan/olap_meta_scan_context.h"
+#include "exec/vectorized/olap_meta_scan_node.h"
+#include "runtime/runtime_state.h"
+
+namespace starrocks {
+
+namespace pipeline {
+
+class OlapMetaChunkSource final : public ChunkSource {
+public:
+    OlapMetaChunkSource(int32_t scan_operator_id, RuntimeProfile* runtime_profile, MorselPtr&& morsel,
+                        vectorized::OlapMetaScanNode* scan_node, OlapMetaScanContextPtr scan_ctx);
+
+    ~OlapMetaChunkSource() override;
+
+    Status prepare(RuntimeState* state) override;
+
+    void close(RuntimeState* state) override;
+
+private:
+    Status _read_chunk(RuntimeState* state, ChunkPtr* chunk) override;
+
+    const workgroup::WorkGroupScanSchedEntity* _scan_sched_entity(const workgroup::WorkGroup* wg) const override;
+
+    vectorized::OlapMetaScanNode* _scan_node;
```

Comment:
avoid using raw pointers?

Reply:
> avoid using raw pointers?

I have removed it because it's not used in OlapMetaChunkSource